src/port1.0/portstartupitem.tcl

```diff
@@ -146,7 +146,7 @@ proc portstartupitem::foreach_startupitem {body} {
                 lappend startupitems_dict($curname) uniquename $uniquename
                 lappend startupitems_dict($curname) plist ${uniquename}.plist
             }
-            lappend startupitems_dict($curname) $key [list $val]
```

Comment:
I don't remember why I used `list` here, but I doubt it was for no reason. It might have been an intermediate stage of development that should have been removed after `list` was added on line 161, or it might have been because of a problem with some other key.

What's really needed here is some tests that check that each `startupitems` key behaves identically to its corresponding `startupitem.*` option with a variety of values (single word, multiple words, characters that need quoting in a list).

Reply:
Your implementation matches what you described in the commit message for 99d5fb30eb4c1ba8b7924807a052291ba31b792c:

```
Example:

startupitem.create  yes
startupitems        name        mydaemon \
                    executable  ${prefix}/bin/mydaemon \
                    name        anotherthing \
                    start       "${prefix}/bin/another --someflag"
                    stop        "${prefix}/bin/another shutdown"
```

I assumed that your implementation was intentional but that you just didn't consider the fact that startupitem.executable supported arguments or that startupitem.start et al supported specifying multiple commands. (I didn't realize that MacPorts supported the latter until I looked at the code two days ago; it's not documented in the guide.)

Adding tests is a good idea... but I've never understood our test suite. Do you know of a specific other test that would be a good model to copy for the startupitems tests, or would you be able to write one test or a placeholder that I could start from?

Reply:
> would you be able to write one test or a placeholder that I could start from?

Sure: 9c74d13

Reply:
Thanks. And would I add the additional tests in the `-body` part of `test startupitems` or should I add additional `test` blocks?

Reply:
How can I run just this test? The [instructions in the guide](https://guide.macports.org/#internals.tests.running) are unclear and nothing I've tried works.

Reply:
> Thanks. And would I add the additional tests in the `-body` part of `test startupitems` or should I add additional `test` blocks?

Ideally each test should test one independent thing. It can make sense to combine multiple related tests into one if the number of tests gets too unwieldy.

> How can I run just this test? The [instructions in the guide](https://guide.macports.org/#internals.tests.running) are unclear and nothing I've tried works.

```
cd src/port1.0
port-tclsh ../tests/test.tcl -t portstartupitem.test
```

Reply:
Thanks. I changed your test a little and added a couple more. Is this a reasonable way to do it? Do we need more tests? (Testing *every* key seems redundant since for example init, start, stop, and restart should behave identically.)